packages/dds/tree/src/test/simple-tree/simpleTree.bench.ts

```diff
@@ -0,0 +1,153 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+
+import { BenchmarkTimer, BenchmarkType, benchmark } from "@fluid-tools/benchmark";
+import {
+	generateDeepSimpleTree,
+	generateWideSimpleTree,
+	readDeepSimpleTree,
+	readWideSimpleTree,
+	writeDeepSimpleTree,
+	writeWideSimpleTree,
+	type DeepTreeNode,
+	type WideTreeNode,
+} from "./utils.js";
+
+// number of nodes in test for wide trees
+const nodesCountWide = [
+	[1, BenchmarkType.Measurement],
+	[100, BenchmarkType.Perspective],
+	[500, BenchmarkType.Measurement],
+];
+// number of nodes in test for deep trees
+const nodesCountDeep = [
+	[1, BenchmarkType.Measurement],
+	[10, BenchmarkType.Perspective],
+	[100, BenchmarkType.Measurement],
+];
+
+describe("SimpleTree benchmarks", () => {
+	describe("Direct SimpleTree", () => {
+		for (const [numberOfNodes, benchmarkType] of nodesCountDeep) {
+			benchmark({
+				type: benchmarkType,
+				title: `Deep Tree as SimpleTree: reads with ${numberOfNodes} nodes`,
+				benchmarkFnCustom: async <T>(state: BenchmarkTimer<T>) => {
```

Comment:
Instead of using benchmarkFnCustom, I think all these tests want to be more like the ones under the "Direct JS Object" ones in packages/dds/tree/src/test/shared-tree/sharedTree.bench.ts, using a `before()` function for the necessary setup, a very simple `benchmarkFn()` that only does the read (or write) and nothing more, and an `after()` function to validate the new state (in the case of writes).

Reply:
Since we are testing the execution time for the simple tree, I think it's a good use case of `benchmarkFnCustom: async <T>(state: BenchmarkTimer<T>)`, which it's the interface have benchmark timer included in Fn. If using `benchmarkFn`, I will need to implement a benchmark timer inside Fn.

Reply:
And since `benchmarkFnCustom` is async, I can't before()
~~~
Read SimpleTree
         @Benchmark @Measurement @ExecutionTime Deep Tree as SimpleTree: reads with 1 nodes:
     TypeError: Cannot read properties of undefined (reading 'node')
      at readDeepSimpleTree (file:///home/tong/workspace/FluidFramework/packages/dds/tree/src/test/simple-tree/utils.ts:87:21)
~~~

Reply:
Updated